Makefile

```diff
@@ -186,3 +187,17 @@ check: ensure-venv
 serve:
 	@echo "The 'serve' target was removed, use 'htmlview' instead" \
 	      "(see https://github.com/python/cpython/issues/80510)"
+
+# TODO make.bat
+include/branches.csv: include/release-cycle.json
+	$(PYTHON) generate-release-cycle.py
+
+include/end-of-life.csv: include/release-cycle.json
+	$(PYTHON) generate-release-cycle.py
+
+include/release-cycle.mmd: include/release-cycle.json
+	$(PYTHON) generate-release-cycle.py
```

Comment:
So this command recreates all three files (guessing from the absence of arguments), but will be called three times when the source file changes?

Reply:
Yes and no: make is smart and will only call the script once. This is the sort of stuff [make](https://www.gnu.org/software/make/manual/make.html#Overview) is built for, it: 

> determines automatically which pieces of a large program need to be recompiled, and issues the commands to recompile them.

To demonstrate by temporarily adding something to the script:

```console
$ git diff
diff --git a/generate-release-cycle.py b/generate-release-cycle.py
index 16dd23b..35bb73c 100644
--- a/generate-release-cycle.py
+++ b/generate-release-cycle.py
@@ -122,6 +122,7 @@ class Versions:
 
 
 def main() -> None:
+    print("main")
     versions = Versions()
     versions.save_csv()
     versions.save_mermaid()
$ touch include/release-cycle.json
$ make versions
python3 generate-release-cycle.py
main
Release cycle data generated.
```

`main` is printed only once.